api/ontology/ontology.go

```diff
@@ -142,3 +143,30 @@ func ResourceIDs(r []IsResource) []string {
 
 	return a
 }
+
+// ProtoResource converrts a [IsResource] to a [Resource]
+func ProtoResource(resource IsResource) *Resource {
+	var (
+		r  Resource
+		m  protoreflect.Message         = r.ProtoReflect()
+		od protoreflect.OneofDescriptor = m.Descriptor().Oneofs().ByName("type")
```

Comment:
```suggestion
		m  = r.ProtoReflect()
		od = m.Descriptor().Oneofs().ByName("type")
```

Reply:
just a suggestion

Reply:
And: I'm not sure if I like to do operations in the var block already. But it is valid Go code so just my two cents